### PR TITLE
Fix bug in order

### PR DESCRIPTION
**Pull request by danhhz**
Fix bug in order by handling with SELECTs and parens

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5439)

### CODE REVIEW

File: sql/testdata/order_by [hunk 1]
```diff
@@ -180,6 +180,14 @@ query I
 3
 2
 
+query I
+(((SELECT a FROM t ORDER BY a DESC LIMIT 4)))
```
Comment (tamird) [inline sql/testdata/order_by]: any significance to the choice of `4` here?

**Comment by tamird** [comment]
LGTM. I'm guessing the limit was not applied at all before?

File: sql/testdata/order_by (hunk 1 above)
Comment (paperstreet) [inline sql/testdata/order_by]: Only that it's less than 5 and greater than 0

**Comment by paperstreet** [comment]
Correct. order and limit were getting dropped on the floor when inside parens

**Comment by RaduBerinde** [comment]
LGTM. Thanks for fixing so quickly. One small comment/request.

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/select.go, line 180 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5439#-KDPQXWFAZoMwf8MXgLC:-KDPQXWFAZoMwf8MXgLD:947385954) ([raw file](https://github.com/cockroachdb/cockroach/blob/c55379fc108c5a00ade25cd2eb634c3be9c62a20/sql/select.go#L180)):*
I found this loop hard to follow at first, but what it's actually doing is quite straightforward.

Could we end the `switch` and `for` here, and then replace the rest with a simple `if selectClause, ok := wrapped.(*SelectClause) { .. } else { .. }` ? That would make it clearer that the loop is only related to `ParenSelect` and the other two are "terminal" cases.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5439#-:-KDPQO3ZGJAL7rLQnhGs:1326540890)*

**Comment by paperstreet** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/select.go, line 180 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5439#-KDPQXWFAZoMwf8MXgLC:-KDPYD7wqFRd6b_CCkas:-2080460893) ([raw file](https://github.com/cockroachdb/cockroach/blob/c55379fc108c5a00ade25cd2eb634c3be9c62a20/sql/select.go#L180)):*
Sure. I kept the "terminal" part as a type switch instead of an if/else since likely there will be more than two cases.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5439)*

**Comment by tamird** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/select.go, line 166 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5439#-KDP_Oqd7IYeU_1ho4CA:-KDP_OqeV8M4aKhjpnep:-531983257) ([raw file](https://github.com/cockroachdb/cockroach/blob/416cf58889a152723da33fb0daf43a7f7c6fa549/sql/select.go#L166)):*
what happened here?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5439)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/select.go, line 166 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5439#-KDP_Oqd7IYeU_1ho4CA:-KDP_j1OkW0KNWvsH7Vb:500116558) ([raw file](https://github.com/cockroachdb/cockroach/blob/416cf58889a152723da33fb0daf43a7f7c6fa549/sql/select.go#L166)):*
oh, this is a `for`, ignore me.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5439)*

**Comment by RaduBerinde** [comment]
:lgtm:

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5439#-:-KDPnaL4O_qEhU_sW6yg:1417088721)*
